### ISSUE TITLE
System Crashes on INSERTS With Missing Columns

### ISSUE THREAD

**Issue by apavlo** [cmu-db/terrier#717]
If you try to insert into a table without specifying values for all the columns, then the server crashes.

**Client**
```
terrier=# CREATE TABLE xxx (id INT, val BIGINT);
CREATE TABLE
terrier=# INSERT INTO xxx VALUES (1);
server closed the connection unexpectedly
        This probably means the server terminated abnormally
        before or while processing the request.
The connection to the server was lost. Attempting reset: Failed.
```

**Server**
```
terrier: /home/pavlo/Documents/Peloton/Github/terrier/src/optimizer/logical_operators.cpp:226: static terrier::optimizer::Operator terrier::optimizer::LogicalInsert::Make(terrier::catalog::db_oid_t, terrier::catalog::namespace_oid_t, terrier::catalog::table_oid_t, std::vector<terrier::common::StrongTypeAlias<terrier::catalog::tags::col_oid_t_typedef_tag, unsigned int> >&&, terrier::common::ManagedPointer<std::vector<std::vector<terrier::common::ManagedPointer<terrier::parser::AbstractExpression> > > >): Assertion `(columns.size() == insert_vals.size()) && ("Mismatched number of columns and values")' failed.
Aborted (core dumped)
```

**Comment by apavlo** [comment]
Note that it crashes even if you specify a default value:

```
terrier=# CREATE TABLE xxx (id INT, val INT DEFAULT 123);
CREATE TABLE
terrier=# INSERT INTO xxx VALUES (1);
server closed the connection unexpectedly
        This probably means the server terminated abnormally
        before or while processing the request.
The connection to the server was lost. Attempting reset: Failed.
```

**Comment by zhou0ricky** [comment]
Problem described above only appears on DEBUG.

On release inserting with or without a default parameter results in adding the tuple into the table with an empty second attribute.

**Comment by lmwnshn** [comment]
Fixed by #727.

**Comment by lmwnshn** [comment]
@zhou0ricky Want to find the JUnit test to re-enable that they mentioned? :)
